### ISSUE TITLE
Reduce EventHandler boilerplate?

### ISSUE THREAD

**Issue by Supreeeme** [not-fl3/egui-miniquad#74]
Currently, integrating with miniquad involves implementing miniquad's EventHandler trait, and delegating a bunch of functions to it. However, I feel like this could be done by egui-miniquad itself through some sort of trait + blanket impl, therefore reducing the amount of boilerplate needed to get up and running.

**Comment by PiotrekB416** [comment]
I don't think this is possible with traits since rust inheritance only guarantees struct with impl a parent trait, which still means you have to implement the methods manually.
Instead you can use a macro to generate the methods in an impl block.
